packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmImpl.kt

```diff
@@ -131,13 +130,14 @@ public class RealmImpl private constructor(
                 }
                 val (frozenReference, fileCreated) = configuration.openRealm(this@RealmImpl)
                 realmFileCreated = assetFileCopied || fileCreated
-                versionTracker.trackAndCloseExpiredReferences(frozenReference)
-                _realmReference.value = frozenReference
+                versionTracker.trackReference(frozenReference)
+                initialRealmReference.value = frozenReference
                 configuration.initializeRealmData(this@RealmImpl, realmFileCreated)
             }
 
             realmScope.launch {
                 notifier.realmChanged().collect {
+                    removeInitialRealmReference()
```

Comment:
It is not enough to closeExpiredReferences() in `removeInitialRealmReference` only the first time that you advance/clear the initial reference, as this might still be referenced. In that case it would not be closed and we will never revisit this and never close it.

Reply:
My bad, yes it is supposed to be triggered on each notification. I am starting to think of adding the GC auto-close.